src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -563,6 +583,9 @@ public static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCall
         public static IAsyncResult BeginGetHostEntry(IPAddress address, AsyncCallback requestCallback, object stateObject)
         {
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostEntry", address);
+
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
I thought about doing that, but these are very cheap, and I like that this way they sort of spell out part of the "contract" of being a public method of this type.